cns/restserver/restserver.go

```diff
@@ -58,7 +59,20 @@ type HTTPRestService struct {
 	state                    *httpRestServiceState
 	podsPendingIPAssignment  *bounded.TimedSet
 	sync.RWMutex
-	dncPartitionKey string
+	dncPartitionKey    string
+	EndpointState      map[string]*EndpointInfo // key : container id
+	EndpointStateStore store.KeyValueStore
+}
+
+type EndpointInfo struct {
+	PodName       string
+	PodNamespace  string
+	IfnameToIPMap map[string]*IPInfo // key : interface name, value : IPInfo
```

Comment:
yeah, I only used pointer because I don't know how big the struct would grow to be. If the struct consistently stay relatively small then I guess we don't have to use pointers.

Reply:
actually, we need pointers for IPInfo during IPconfig request and release, as we don't want to create a new copy of the arrays within IPInfo, update with new entries, and then reassign it every time.